### PR TITLE
Retry on HttpError

### PR DESCRIPTION
**Pull request by pboothe**
This change is [Reviewable](https://reviewable.io/reviews/m-lab/scraper/55)

### CODE REVIEW

**Comment by pboothe** [comment]
Fixes a problem in prod:

```
[2017-06-01 18:31:30,816 INFO discovery.py:863 rsync://ndt.iupui.mlab1.lhr01.measurement-lab.org:7999/ndt] URL being requested: POST https://www.googleapis.com/upload/storage/v1/b/scraper-mlab-oti/o?uploadType=resumable&alt=json&name=ndt%2F2017%2F05%2F25%2F20170525T000000Z-mlab1-lhr01-ndt-0204.tgz
Traceback (most recent call last):
  File "/run_scraper.py", line 207, in <module>
    main(sys.argv)
  File "/run_scraper.py", line 177, in main
    exceptions=scraper.RecoverableScraperException)
  File "/usr/local/lib/python2.7/dist-packages/retry/api.py", line 101, in retry_call
    return __retry_internal(partial(f, *args, **kwargs), exceptions, tries, delay, max_delay, backoff, jitter, logger)
  File "/usr/local/lib/python2.7/dist-packages/retry/api.py", line 33, in __retry_internal
    return f()
  File "/scraper.py", line 769, in upload_stale_disk
    last_okay_day)
  File "/scraper.py", line 792, in upload_up_to_date
    args.rsync_module, args.bucket)
  File "<decorator-gen-3>", line 2, in upload_tarfile
  File "/usr/local/lib/python2.7/dist-packages/prometheus_client/core.py", line 879, in wrapped
    return func(*args, **kwargs)
  File "/scraper.py", line 526, in upload_tarfile
    progress, response = request.next_chunk()
  File "/usr/local/lib/python2.7/dist-packages/oauth2client/_helpers.py", line 133, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/googleapiclient/http.py", line 979, in next_chunk
    return self._process_response(resp, content)
  File "/usr/local/lib/python2.7/dist-packages/googleapiclient/http.py", line 1010, in _process_response
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 503 when requesting https://www.googleapis.com/upload/storage/v1/b/scraper-mlab-oti/o?uploadType=resumable&alt=json&name=ndt%2F2017%2F05%2F25%2F20170525T000000Z-mlab1-lhr01-ndt-0204.tgz returned "Service Unavailable">
```

File: scraper.py [hunk 1]
```diff
@@ -495,22 +496,28 @@ def create_temporary_tarfiles(tar_binary, tarfile_template, directory, day,
...
+             logger=logging.getLogger())
 def upload_tarfile(service, tgz_filename, date, experiment,
                    bucket):  # pragma: no cover
     """Uploads a tarfile to Google Cloud Storage for later processing.
 
-    Puts the file into a GCS bucket. If a file of that same name already
-    exists, the file is overwritten.  If the upload fails, HttpError is raised.
+    Puts the file into a GCS bucket. If a file of that same name already exists,
+    the file is overwritten.  If the upload fails, the upload is retried until
+    it succeeds, although we perform exponential backoff with a maximum wait
```
Comment (stephen-soltesz) [inline scraper.py]: "If the upload fails, the upload is retried until it succeeds"

Is there a way to limit this to only 500 errors?

Comment (pboothe) [inline scraper.py]: PTAL

**Comment by stephen-soltesz** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/scraper/55#-:-Kl_HHreXBspZI6VI7Nk:bnfp4nl)*

**Pull request merged**